src/features/pb.rs

```diff
@@ -25,5 +25,19 @@ pub mod user {
 pub mod chart {
     //! Contains chart protobufs
 
+    use convert_case::{Case, Casing};
+
     include!("../../proto/ratings.features.chart.rs");
+
+    impl Category {
+        /// Allows our enum to be represented in snake case for consistency with what `snapd`
+        /// returns, unfortunately this allocates but it sucks less than manually
+        /// maintaining an enum that returns a &'static str for all of them.
+        ///
+        /// Sadly, since it's a protobuf we can't really use any of the derive
+        /// crates to make a "cheaper" version so this it is.
```

Comment:
Don't worry about optimisation at this point ... totally fine to be less efficient (but nto glaringly so!). If performance becomes and issue then we can measure and address bottlenecks then.